Signal/src/ViewControllers/ProfileViewController.m

```diff
@@ -285,6 +387,50 @@ - (void)updateAvatarView
 
 #pragma mark - AvatarViewHelperDelegate
 
++ (BOOL)shouldDisplayProfileViewOnLaunch
+{
+    // Only nag until the user sets a profile _name_.  Profile names are
+    // recommended; profile avatars are optional.
+    if ([OWSProfileManager sharedManager].localProfileName.length > 0) {
+        return NO;
+    }
+
+    NSTimeInterval kProfileNagFrequency = kDayInterval * 30;
+    NSDate *_Nullable lastPresentedDate =
+        [[[TSStorageManager sharedManager] dbReadConnection] dateForKey:kProfileView_LastPresentedDate
```

Comment:
My point was not about if it will compile, but to double down on what I believe to be best practice when working with yap DB connections. Specifically - a concurrency concern (in this case, coherent read/write to when the nag was last presented) should be put through a single dbConnection.